environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java

```diff
@@ -73,6 +75,25 @@ public ValidationResultBuilder validateNetworkCreation(Environment environment,
         return networkCreationValidator.validateNetworkCreation(environment, network, subnetMetas);
     }
 
+    public ValidationResult validateParentChildRelation(Environment environment, String parentEnvironmentCrn) {
+        ValidationResultBuilder resultBuilder = new ValidationResultBuilder();
+
+        resultBuilder.ifError(() -> Objects.nonNull(parentEnvironmentCrn) && Objects.isNull(environment.getParentEnvironment()),
+                String.format("Active parent environment with crn '%s' is not available in account '%s'.", parentEnvironmentCrn, environment.getAccountId()));
+        if (Objects.nonNull(environment.getParentEnvironment())) {
+            resultBuilder.ifError(() -> environment.getParentEnvironment().getStatus() != EnvironmentStatus.AVAILABLE,
+                    "Parent environment should be in 'AVAILABLE' status.");
+            resultBuilder.ifError(() -> Objects.nonNull(environment.getParentEnvironment().getParentEnvironment()),
+                    "Parent environment is already a child environment.");
+            resultBuilder.ifError(() -> !CloudPlatform.YARN.name().equalsIgnoreCase(environment.getCloudPlatform()),
```

Comment:
can we expose the supported platforms to an application property ?

Reply:
Done in af20f701d8c3e87addc57f612c9b6fc642d0daf5